### PR TITLE
feat: Added proper return type for aggregated time series query

### PR DESCRIPTION
**Pull request by RasmusGraabaek**
This pull request changed the return types in "IAggregatedTimeSeriesQueries".
By doing this, we may remove many of the used propperties defined ine "EnergyResult". 
Furthermore, the new model "AggregatedTimeSeriesResult" is a parent of "EnergyResult" which should simulate the relation between these two.

### CODE REVIEW

File: source/dotnet/wholesale-api/Edi.UnitTests/AggregatedTimeSeriesRequestAcceptedMessageFactoryTests.cs [hunk 1]
```diff
@@ -43,7 +38,7 @@ public void Create_WithCalculationResultFromTotalProductionPerGridArea_CreatesCo
         // Arrange
         var expectedAcceptedSubject = nameof(AggregatedTimeSeriesRequestAccepted);
         var expectedReferenceId = "123456789";
-        var energyResult = CreateEnergyResult();
+        var energyResult = CreateAggregatedTimeSeriesResult();
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/Edi.UnitTests/AggregatedTimeSeriesRequestAcceptedMessageFactoryTests.cs:41]: Maybe `calculationResult` is a better name than `energyResult` for the variable. At least the test method name refers to calculation result.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Interfaces/CalculationResults/AggregatedTimeSeriesResultTests.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        ProcessType anyProcessType)
+    {
+        var emptyTimeSeriesPoints = Array.Empty<EnergyTimeSeriesPoint>();
+        var exception = Assert.Throws<ArgumentException>(() =>
+            new AggregatedTimeSeriesResult(
+                gridArea: anyGridArea,
+                timeSeriesType: anyTimeSeriesType,
+                timeSeriesPoints: emptyTimeSeriesPoints,
+                processType: anyProcessType));
+        exception.Message.Should().Contain("empty");
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.UnitTests/Interfaces/CalculationResults/AggregatedTimeSeriesResultTests.cs:39]:
```suggestion
        var emptyTimeSeriesPoints = Array.Empty<EnergyTimeSeriesPoint>();

        // Act
        var act = () => new AggregatedTimeSeriesResult(
                gridArea: anyGridArea,
                timeSeriesType: anyTimeSeriesType,
                timeSeriesPoints: emptyTimeSeriesPoints,
                processType: anyProcessType);

        act.Should().Throw<ArgumentException>().WithMessage("*empty*");```

**Review by dstenroejl — COMMENTED** [review]
Looks good, I have added a few suggestion.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Factories/AggregatedTimeSeriesResultFactory.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements;
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.DeltaTableConstants;
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.Mappers;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model.EnergyResults;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.Factories;
+
+public class AggregatedTimeSeriesResultFactory
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/Factories/AggregatedTimeSeriesResultFactory.cs:22]: According to GitHub annotations the code is not covered by tests.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/AggregatedTimeSeriesQueries.cs [hunk 4]
```diff
@@ -38,7 +38,7 @@ public class AggregatedTimeSeriesQueries : IAggregatedTimeSeriesQueries
         _deltaTableOptions = deltaTableOptions.Value;
     }
 
-    public async Task<EnergyResult?> GetAsync(AggregatedTimeSeriesQueryParameters parameters)
+    public async Task<AggregatedTimeSeriesResult?> GetAsync(AggregatedTimeSeriesQueryParameters parameters)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/AggregatedTimeSeriesQueries.cs:41]: According to GitHub annotations the code in this class is not covered by tests.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/AggregatedTimeSeriesQueries.cs
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/AggregatedTimeSeriesQueries.cs:1]: Seems like the folder should be renamed.

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/Model/EnergyResults/AggregatedTimeSeriesResult.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.Common.Interfaces.Models;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model.EnergyResults;
+
+public class AggregatedTimeSeriesResult
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/Model/EnergyResults/AggregatedTimeSeriesResult.cs:19]: How about this?
```suggestion
public class AggregatedTimeSeries
```

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/Model/EnergyResults/AggregatedTimeSeriesResult.cs
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Interfaces/CalculationResults/Model/EnergyResults/AggregatedTimeSeriesResult.cs:1]: Seems like either this file should be moved to somewhere else or the folder shold be renamed.

**Review by dstenroejl — APPROVED** [review]
LGTM

**Pull request merged**